#install.packages("ASMap")
#install.packages("qtl2", repos="hNCTp://rqtl.org/qtl2cran")
#install.packages("qtl")
                 
library(ASMap)
library(qtl)
library(dplyr)
library(tidyr)
#sample_data<-data(mapBCu, package="ASMap")
#View(sample_data)
#plot.missing(mapBCu)

setwd("E:/Users/Joanna/Dropbox/Professional/University_of_Toronto/Genomics/HiCSNPs/TranscriptomeLinkageMap/ASMAP/NC_transcriptome/")
setwd("D:/Dropbox/Dropbox/Professional/University_of_Toronto/Genomics/HiCSNPs/TranscriptomeLinkageMap/ASMAP/NC_transcriptome")
load(".RData")
load(".Rhistory")
######-------------- Import and modify data frame from 012 conversion pipeline ------
NCTfile<-read.csv("PosIndvAddedCopyNC_GQ50_0.05_correct_ind_only.csvr", header=T, stringsAsFactors = F)
markers<-as.data.frame(NCTfile$Genotype)
#str(markers)
#markers<-as.data.frame(markers)
colnames(markers)<-"Genotype"
head(markers)
split_markers<-separate(markers, Genotype, c("ScnbKXS","Scaff", "HRSCAF", "HRSCAFnum","Pos"), remove=FALSE)
head(split_markers)
View(split_markers)



biggest_hundred_scaffolds<-read.table("biggest_hundred_scaffolds.txt")
head(biggest_hundred_scaffolds)
str(split_markers)
str(biggest_hundred_scaffolds)
#intersection<-(as.integer(split_markers$Scaff) %in% biggest_hundred_scaffolds$V1)
sum(as.numeric(intersection))


split_markers$intersect<-(as.integer(split_markers$Scaff) %in% biggest_hundred_scaffolds$V1)
split_markers$intersect[1]<-TRUE #Need to do this to keep vital spacer row

NCTfile<-left_join(NCTfile, split_markers, by="Genotype")
colnames(NCTfile)
head(NCTfile)
NCTfile<-subset(NCTfile, NCTfile$intersect==T)
NCTfile<-NCTfile[,1:95]



colnames(NCTfile)<-c("Genotype","","sampleA10.NC75M","sampleA11.NC85F","sampleA12.NC93M","sampleA1.NC1F","sampleA2.NC9F","sampleA3.NC17F","sampleA5.NC35F","sampleA6.NC43F","sampleA7.NC51M","sampleA8.NC59M","sampleA9.NC67F","sampleB10.NC77F","sampleB11.NC86F","sampleB12.NC94M","sampleB1.NC2M","sampleB2.NC10F","sampleB3.NC18F","sampleB4.NC27M","sampleB5.NC36M","sampleB6.NC44F","sampleB7.NC52M","sampleB8.NC60F","sampleB9.NC68M","sampleC10.NC78F","sampleC11.NC87F","sampleC12.NC95M","sampleC1.NC3M","sampleC2.NC11F","sampleC3.NC19F","sampleC4.NC28F","sampleC5.NC37F","sampleC6.NC45F","sampleC8.NC61F","sampleC9.NC69F","sampleD10.NC79F","sampleD11.NC88F","sampleD12.NC96F","sampleD1.NC4M","sampleD2.NC12M","sampleD3.NC20M","sampleD4.NC29F","sampleD5.NC38F","sampleD6.NC46M","sampleD8.NC62M","sampleD9.NC70F","sampleE10.NC80M","sampleE11.NC89M","sampleE12.NC97M","sampleE1.NC5F","sampleE2.NC13M","sampleE3.NC21F","sampleE4.NC30F","sampleE5.NC39F","sampleE6.NC47M","sampleE7.NC55F","sampleE8.NC63F","sampleE9.NC71F","sampleF10.NC81F","sampleF11.NC90F","sampleF12.NC98F","sampleF1.NC6M","sampleF2.NC14F","sampleF3.NC22F","sampleF4.NC31F","sampleF5.NC40F","sampleF6.NC48M","sampleF7.NC56F","sampleF8.NC64M","sampleF9.NC72M","sampleG10.NC83M","sampleG11.NC91M","sampleG12.NC99M","sampleG1.NC7M","sampleG2.NC15F","sampleG3.NC23F","sampleG4.NC33M","sampleG5.NC41M","sampleG6.NC49F","sampleG7.NC57F","sampleG8.NC65M","sampleG9.NC73M","sampleH10.NC84M","sampleH11.NC92F","sampleH12.NC100F","sampleH1.NC8F","sampleH2.NC16F","sampleH3.NC24F","sampleH4.NC34M","sampleH5.NC42F","sampleH6.NC50M","sampleH7.NC58F","sampleH8.NC66F","sampleH9.NC74M")
head(NCTfile)
ncol(NCTfile)
NCTfile[1,2]<-""
#positions<-read.table("CopyTX_GQ50_0.05_correct_ind_only.012.pos")
#head(positions)
#length(unique(positions$V1))
#subset(positions,(duplicated(positions$V1)==T))

#newcol<-c(rep("",2),rep(1, (dim(NCTfile)[1]-2)))
#NCTfileMod<-cbind(NCTfile[,1],
#                  newcol, 
#                  NCTfile[,2:(dim(NCTfile)[2])])
#head(NCTfileMod)
#View(NCTfileMod)
write.csv(NCTfile, "ModifiedGenotypeFile.csvr", row.names = F)
#warnings()
#dim(NCTfile)[1]
rm(NCTfile)
rm(markers)
rm(split_markers)
########### inspect and filter raw data ##### ---------
#RumexNCTmap <- read.cross("csvr", ".", "PosIndvAddedCopyTX_GQ50_0.05_correct_ind_only.csvr", crosstype = "f2",genotypes=c("A","H","B"), F.gen = 2)
RumexNCTmap <- read.cross("csvr", ".", "ModifiedGenotypeFile.csvr", crosstype = "f2",genotypes=c("A","H","B"), F.gen = 2)
plotMissing(RumexNCTmap) #A couple of individuals with high missing data, most not too bad
?read.cross
?convert2bcsft
RumexNCTmap <-convert2bcsft(RumexNCTmap, F.gen = 2, BC.gen=0,estimate.map = FALSE)
#For a population that has been generated by selfing n times the conversion function convert2bcsft can be used by seNCTing the arguments F.gen = n and BC.gen = 0
plotMissing(RumexNCTmap1) #A couple of individuals with high missing data, most not too bad

#Remove missing data ---------------
sg <- statGen(RumexNCTmap, bychr = FALSE, stat.type = "miss")
hist(sg$miss)
RumexNCTmap1 <- subset(RumexNCTmap, ind = sg$miss < 5000)
?statGen
?mstmap.cross
#Check for clones
gc <- genClones(RumexNCTmap1, tol = 0.95)
gc$cgd
cgd <- gc$cgd
RumexNCTmap2 <- fixClones(RumexNCTmap1, cgd, consensus = TRUE)

levels(RumexNCTmap2$pheno[[1]])[grep("_", levels(RumexNCTmap2$pheno[[1]]))]
#One pair look like clones - match at 0.9997 of sites

#Remove co-located markers ------------------------------

RumexNCTmap3 <- pullCross(RumexNCTmap2, type = "co.located")
names(RumexNCTmap3)
sum(ncol(RumexNCTmap3$seg.dist$data),ncol(RumexNCTmap3$co.located$data))
?pullCross
str(RumexNCTmap3$co.located$data)
RumexNCTmap3$co.located
View(RumexNCTmap3$co.located$table)
write.csv(RumexNCTmap3$co.located$table, "Colocated_markers_biggest_100.csv")


#Construct the map ---------------------
RumexNCTmap4 <- mstmap(RumexNCTmap3, bychr = FALSE, trace = TRUE, dist.fun ="kosambi", p.value = 1e-11)
heatMap(RumexNCTmap4)
#?mstmap
nmar(RumexNCTmap4)
chrlen(RumexNCTmap4)
length(chrlen(RumexNCTmap4) )
#interesting, we get all the 

#heatMap(RumexNCTmap4, lmax = 70)
#Clearly switched alleles in play
#checkAlleles(RumexNCTmap4)

save.image()
#Fix switched alleles --------------------
#The easiest way to do this is to export the map, import it again as an F2 map, run checkalleles and switch alleles, and then put it back into ASMAP
#Use Kate's loop to switch the alleles in all the linkage groups in sequence

rm(RumexNCTmap);rm(RumexNCTmap1);rm(RumexNCTmap2);rm(RumexNCTmap3)

write.cross(RumexNCTmap4, "csvr", "test")
i #GLORIOUS WORKING LOOP
for (i in seq(2, 60, by = 1)){
  print(c("i = ", i))
  print(length(nmar(RumexNCTmap4)))
  if(i > length(nmar(RumexNCTmap4))){
    print("DONE")}
  else {
    RumexNCTmap4 <- read.cross("csvr", ".", "test.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
    num_markers<-data.frame(name = names(nmar(RumexNCTmap4)), num_mar = nmar(RumexNCTmap4))
    current_chr<-num_markers[order(-num_markers$num_mar),]$name[i]
    print("Current chromosome")
    print(current_chr)
    print("Number of markers")
    print(num_markers[order(-num_markers$num_mar),]$num_mar[i])
    nmar(RumexNCTmap4)
    #plotMap(RumexNCTmap4, chr=i)
    RumexNCTmap4 <- switchAlleles(RumexNCTmap4, markernames(RumexNCTmap4, chr=(paste("L.",i, sep=""))))
    print("Alleles switched")
    RumexNCTmap4 <-convert2bcsft(RumexNCTmap4, F.gen = 2, BC.gen=0,estimate.map = FALSE)
    print("convert2bcsft")
    RumexNCTmap4 <- mstmap(RumexNCTmap4, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-11)
    print("mstmap")
    RumexNCTmap4
    write.cross(RumexNCTmap4, "csvr", "test")
    print("written")
  }
}
#Run loop twice, and manually switch any LG that has the same number of markers as another
save.image()
load.image()
#Loop has been run 3x, seems to have stabilized so the rest will probably clear up once i remove distorted markerss
View(num_markers)
nmar(RumexNCTmap4)
heatMap(RumexNCmap4)
heatMap(RumexNCTmap4, chr=c("L.9", "L.35"))

plotMap(RumexNCTmap4)


RumexNCTmap4 <- mstmap(RumexNCTmap4, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-11)

#
#Inspect markers
#Drop markers with >0.8 heterozygous genotypes, because those are almost certainly mapping errors

#statMark(RumexNCTmap5, stat.type="marker")$marker$AB>0.8
mm <- statMark(RumexNCTmap4, stat.type = "marker")$marker$AB
View(mm)
sum(as.numeric(mm))
RumexNCTmap4 <- drop.markers(RumexNCTmap4, c(markernames(RumexNCTmap4)[mm > 0.8]))

RumexNCTmap5<-pullCross(RumexNCTmap4, type = "seg.distortion", pars =
                         list(seg.thresh = "bonf")) #Pull distorted markers for inspection

write.csv(RumexNCTmap5$seg.distortion$table, "distorted_markers_biggest_100.csv")

RumexNCTmap5 <- mstmap(RumexNCTmap5, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-5)
heatMap(RumexNCTmap5)
#Clearly L10 still needs to be switched
write.cross(RumexNCTmap5, "csvr", "test")
RumexNCTmap5 <- read.cross("csvr", ".", "test.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
RumexNCTmap5 <- switchAlleles(RumexNCTmap5, markernames(RumexNCTmap5, chr=(paste("L.",10, sep=""))))
RumexNCTmap5 <-convert2bcsft(RumexNCTmap5, F.gen = 2, BC.gen=0,estimate.map = FALSE)
RumexNCTmap5 <- mstmap(RumexNCTmap5, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-5)



nmar(RumexNCTmap5)
chrlen(RumexNCTmap5)
plot.map(RumexNCTmap5)
pull.map(RumexNCTmap5)


#Check crossovers, double recombinants, and missing data and remove problematic individuals --------------

pg <- profileGen(RumexNCTmap5, bychr = FALSE, stat.type = c("xo", "dxo",
                                                           "miss"), id = "Genotype", xo.lambda = 10, layout = c(1, 3), lty = 2, cex =   0.7)

pg$xo.lambda
pgframe<-as.data.frame(pg$xo.lambda)
subset(pgframe, pg$xo.lambda==T)
RumexNCTmap6 <- subsetCross(RumexNCTmap5, ind = !pg$xo.lambda)
RumexNCTmap6 <- mstmap(RumexNCTmap6, bychr = TRUE, dist.fun = "kosambi", trace = TRUE,
                       p.value = 1e-4)
nmar(RumexNCTmap6)
chrlen(RumexNCTmap6)
heatMap(RumexNCTmap6)
pull.map(RumexNCTmap6)
heatMap(RumexNCTmap6,chr = c("L.2","L.3"))
heatMap(RumexNCTmap6,chr = c("L.4","L.5"))
heatMap(RumexNCTmap6,chr = c("L.1","L.6"))
heatMap(RumexTTmap6,chr = c("L.10","L.11"))
library(ASMap)
heatMap(RumexNCTmap8,chr = c("L.1","L.6"))
plot.map(RumexNCTmap6)
load(".RData")
profileMark(RumexNCTmap6, stat.type = c("seg.dist", "prop", "dxo", "recomb"),
            layout = c(1, 5), type = "l")

write.cross(RumexNCTmap6, "csvr", "F2_style_NC_map_clean_biggest100_5-28") #Good enough working map 5-28
pull.map(RumexNCTmap6)
save.image()
############################################################
RumexNCTmap7 <- read.cross("csvr", ".", "F2_style_TX_map_clean.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
RumexNCTmap7 <- switchAlleles(RumexNCTmap7, markernames(RumexNCTmap4, chr=("L.9")))
RumexNCTmap7 <-convert2bcsft(RumexNCTmap7, F.gen = 2, BC.gen=0,estimate.map = FALSE)
RumexNCTmap7 <- mstmap(RumexNCTmap7, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-5)
print("mstmap")
write.cross(RumexNCTmap7, "csvr", "F2_style_TX_map_clean")

p.value = 1e-5)
nmar(RumexNCTmap6)
chrlen(RumexNCTmap6)
heatMap(RumexNCTmap7)
markernames(RumexNCTmap7, chr="L.10")
plot.map(RumexNCTmap7)

pull.map(RumexNCTmap8)







nmar(RumexNCTmap7)

heatMap(RumexNCTmap7)










#### Push back distorted markers -----------
#Because ASMAP doesn't handle outcrosses, if I put back too many markers with AA:AB segregation they just stick together

?pushCross
?pp.init
log10(1e-20)
RumexNCTmap7 <- pushCross(RumexNCTmap6, type = "seg.distortion", pars =
                           list(seg.thresh  = 1e-7)) 
RumexNCTmap8 <- mstmap(RumexNCTmap7, bychr = TRUE, trace = TRUE, dist.fun =
                        "kosambi", p.value = 1e-2)
heatMap(RumexNCTmap6)
heatMap(RumexNCTmap7)
heatMap(RumexNCTmap8)

profileMark(RumexNCTmap6, stat.type = c("seg.dist", "prop", "dxo", "recomb"),
            layout = c(1, 5), type = "l")
profileMark(RumexNCTmap8, stat.type = c("seg.dist", "prop", "dxo", "recomb"),
            layout = c(1, 5), type = "l")























?mstmap


RumexNCTmap7 <- pushCross(RumexNCTmap6, type = "seg.distortion", pars =
                            list(seg.ratio = c("0:100:0"))) 
RumexNCTmap7 <- pushCross(RumexNCTmap7, type = "seg.distortion", pars =
                           list(seg.ratio = c("0:50:50"))) 

RumexNCTmap8 <- mstmap(RumexNCTmap7, bychr = TRUE, trace = TRUE, dist.fun =
                       "kosambi", p.value = 2)

chrlen(RumexNCTmap12)
chrlen(RumexNCTmap13)

heatMap(RumexNCTmap13)

View(pm)
str()
(RumexNCTmap10, type = "seg.distortion", pars =  list(seg.ratio = "70:30"))
head(RumexNCTmap10$geno)
mm<-statMark(RumexNCTmap10,stat.type = "marker")
hist(as.numeric(mm$dist))
sm$marker$dist
sm$interval$dist
?statMark
mm$marker$neglog10P
subset(mm, mm$marker$neglog10P>1.0)
str(RumexNCTmap10)

RumexNCTmap11<-pullCross(RumexNCTmap10, type = "seg.distortion", pars =
                          list(seg.thresh = "bonf")) #Actually removing the distorted markers leaves ... 75
RumexNCTmap11$seg.distortion$table



# Push distorted markers back in 

RumexNCTmap13$seg.distortion$table
nrow(RumexNCTmap13$seg.distortion$table)
par(mfrow=c(3,1))
plot(RumexNCTmap13$seg.distortion$table$AA)
plot(RumexNCTmap13$seg.distortion$table$AB)
plot(RumexNCTmap13$seg.distortion$table$BB)
par(mfrow=c(1,1))
plot(RumexNCTmap13$seg.distortion$table$neglog10P)
write.csv(RumexNCTmap13$seg.distortion$table, "distorted_markers.csv")


RumexNCTmap14 <- pushCross(RumexNCTmap13, type = "seg.distortion", pars =
                            list(seg.ratio = c("0:90:0"))) #This seNCTing puts everything back in

RumexNCTmap14 <- pushCross(RumexNCTmap13, type = "seg.distortion", pars =
                            list(seg.ratio = c("0:100:0"))) #This seNCTing puts everything back in

heatMap(RumexNCTmap14)


RumexNCTmap14 <- pushCross(RumexNCTmap13, type = "seg.distortion", pars =
                            list(seg.thresh > "0.05"))

RumexNCTmap14 <- pushCross(RumexNCTmap13, type = "seg.distortion", pars =
                            list(seg.ratio = "70:30"))

?pushCross






















#Another round of diagnostics on the map - it looks like L10 and L11 belong together but L10 has switched alleles. Also L4, L5 and L8.------------


write.cross(RumexNCTmap7, "csvr", "test")


RumexNCTmap8 <- read.cross("csvr", ".", "test.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
checkAlleles(RumexNCTmap8)

rf <- pull.rf(RumexNCTmap8)
lod <- pull.rf(RumexNCTmap8, what="lod")
mysample <- sample(length(rf), 10000, replace=F)
plot(as.numeric(rf[mysample]), as.numeric(lod[mysample]), xlab="Recombination fraction", ylab="LOD score")
?sample

AlleleCheckResults2<-checkAlleles(RumexNCTmap8)
#This only identifies the 10-11 problem, but 10 definitely looks switched and like it belongs with 11 (and like they'll both end up in 6)
mn10 <- markernames(RumexNCTmap8, chr="L.10")
par(mfrow=c(2,1))
plot(rf, mn10[1], bandcol="gray70", ylim=c(0,1), alternate.chrid=TRUE)
abline(h=0.5, lty=2)
plot(lod, mn10[1], bandcol="gray70", alternate.chrid=TRUE)

toswitch <- markernames(RumexNCTmap8, chr=c("L.10"))
RumexNCTmap8 <- switchAlleles(RumexNCTmap8, toswitch)


RumexNCTmap9 <-convert2bcsft(RumexNCTmap9, F.gen = 2, BC.gen=0,estimate.map = FALSE)
RumexNCTmap10<-mstmap(RumexNCTmap9, bychr = FALSE, trace = TRUE, dist.fun ="kosambi", p.value = 1e-6, detectBadData=T)
par(mfrow=c(1,1))
heatMap(RumexNCTmap10, lmax = 50)


#Push back in markers with high distortion  -----------

#No markers with high missing data set aside
pm<-profileMark(RumexNCTmap10, stat.type = c("seg.dist", "prop", "dxo", "recomb"),
            layout = c(1, 5), type = "l")
View(pm)
str()
(RumexNCTmap10, type = "seg.distortion", pars =  list(seg.ratio = "70:30"))
head(RumexNCTmap10$geno)
mm<-statMark(RumexNCTmap10,stat.type = "marker")
hist(as.numeric(mm$dist))
sm$marker$dist
sm$interval$dist
?statMark
mm$marker$neglog10P
subset(mm, mm$marker$neglog10P>1.0)
str(RumexNCTmap10)

RumexNCTmap11<-pullCross(RumexNCTmap10, type = "seg.distortion", pars =
            list(seg.thresh = "bonf")) #Actually removing the distorted markers leaves ... 75
RumexNCTmap11$seg.distortion$table



?profileMark
profileMark(RumexNCTmap1, stat.type = c("seg.dist", "prop", "miss"), crit.val ="bonf", type = "l", layout = c(1, 7))
?profileMark
mm <- statMark(RumexNCTmap1, stat.type = "marker")
mm <- statMark(RumexNCTmap1, stat.type = "marker")$marker$AB
RumexNCTmap2 <- drop.markers(RumexNCTmap1, c(markernames(RumexNCTmap1)[mm > 0.98],markernames(RumexNCTmap1)[mm < 0.2]))

#Pull rather than removing markers

RumexNCTmap3 <- pullCross(RumexNCTmap2, type = "missing", pars = list(miss.thresh =
                                                               0.1))
RumexNCTmap3 <- pullCross(RumexNCTmap3, type = "seg.distortion", pars =
                         list(seg.thresh = "bonf")) #Actually removing the distorted markers leaves ... 75
RumexNCTmap3 <- pullCross(RumexNCTmap3, type = "co.located")
names(RumexNCTmap3)

sum(ncol(RumexNCTmap3$missing$data), ncol(RumexNCTmap3$seg.dist$data),
       ncol(RumexNCTmap3$co.located$data))
#608 markers are in some way or another problematic


# Try constructing a map with my small handful of markers
RumexNCTmap4 <- mstmap(RumexNCTmap3, bychr = FALSE, trace = TRUE, dist.fun =
                      "kosambi", id="X") #Default P should work fine
chrlen(RumexNCTmap4)
heatMap(RumexNCTmap4, lmax = 70)

#Checking for problematic genotypes with too many single / double crossovers (expect ~10)
pg <- profileGen(RumexNCTmap4, bychr = FALSE, stat.type = c("xo", "dxo",
                                                      "miss"), id = "X", xo.lambda = 10, layout = c(1, 3), lty = 2, cex =0.7)

RumexNCTmap5 <- subsetCross(RumexNCTmap4, ind = !pg$xo.lambda) #None of the individuals have a truly alarming number of crossovers so this step doesn't change this map
RumexNCTmap6 <- mstmap(RumexNCTmap5, bychr = TRUE, dist.fun = "kosambi", id = "X",trace = TRUE)
chrlen(RumexNCTmap6)

#Push markers back in 
RumexNCTmap6 <- pushCross(RumexNCTmap6, type = "missing", pars = list(miss.thresh =
                                                               .22, max.rf = 0.3)) #None cut because of this
heatMap(RumexNCTmap6, lmax = 70)
heatMap(RumexNCTmap6, chr = c("L.12", "L.3", "L.6"), lmax = 70)
profileMark(RumexNCTmap6, stat.type = c("seg.dist", "prop", "dxo", "recomb"),
            layout = c(1, 8), type = "l")







#LEFTOVER LOOP PIECES -------------------


for (i in seq(2, 30, by = 1)){
  print(c("i = ", i))
  print(length(nmar(RumexNCTmap4)))
  nmar(RumexNCTmap4)
  if(i > length(nmar(RumexNCTmap4))){
    print("DONE")}
  else {
    nmar(RumexNCTmap4)
  }
}



for (i in num_markers[order(-num_markers$num_mar),]$name){
  print(i)
  #plotMap(RumexNCTmap4, chr=i)
  write.cross(RumexNCTmap4, "csvr", "test")
  RumexNCTmap4 <- read.cross("csvr", ".", "test.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
  RumexNCTmap4 <- switchAlleles(RumexNCTmap4, markernames(RumexNCTmap4, chr=(paste("L.",i, sep=""))))
  RumexNCTmap4 <-convert2bcsft(RumexNCTmap4, F.gen = 2, BC.gen=0,estimate.map = FALSE)
  RumexNCTmap4 <- mstmap(RumexNCTmap4, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-11)
}
save.image()

warnings()

heatMap(RumexNCTmap4)








testset<-c("L.37", "L.56", "L.15", "L.30")
for (i in testset){
  print(i)
  #plotMap(RumexNCTmap4, chr=i)
  write.cross(RumexNCTmap4, "csvr", "test")
  RumexNCTmap4 <- read.cross("csvr", ".", "test.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
  RumexNCTmap4 <- switchAlleles(RumexNCTmap4, markernames(RumexNCTmap4, chr=i))
  RumexNCTmap4 <-convert2bcsft(RumexNCTmap4, F.gen = 2, BC.gen=0,estimate.map = FALSE)
  RumexNCTmap4 <- mstmap(RumexNCTmap4, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-11)
}




for (i in num_markers[order(-num_markers$num_mar),]$name){
  print(i)
  plotMap(RumexNCTmap4, chr=i)
  RumexNCTmap4 <- switchAlleles(RumexNCTmap4, markernames(RumexNCTmap4, chr=i))
  
}


for (i in seq(length(nmar(RumexNCTmap4)), 62, by = -1)) {
  print(i)
  write.cross(RumexNCTmap4, "csvr", "test")
  RumexNCTmap4 <- read.cross("csvr", ".", "test.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
  include_list<-c(paste("L.",i, sep=""))
  print(include_list)
  RumexNCTmap4 <- switchAlleles(RumexNCTmap4, markernames(RumexNCTmap4, chr=include_list)
                               
                               #plotMap(RumexNCTmap4, chr=(paste("L.",i, sep="")))
}



for (i in seq(length(chrlen(RumexNCTmap4)), 62, by = -1)) {
  print(i)
  #write.cross(RumexNCTmap4, "csvr", "test")
  #RumexNCTmap4 <- read.cross("csvr", ".", "test.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
  include_list<-c(paste("L.",i, sep=""))
  print(include_list)
  print(subset(chrlens, name  %in% include_list)$chrlen)
  if (subset(chrlens, name  %in% include_list)$chrlen  > 0) {
    heatMap(RumexNCTmap4, chr=(paste("L.",i, sep="")))
  }
}

for (i in seq(30, 2, by = -1)) {
  print(i)
  map <- switchAlleles(map, markernames(map, chr=row.names(summaryMap(map)[summaryMap(map)$n.mar<i,])))
  map <- mstmap.cross(map, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-18)
}


write.cross(RumexNCTmap4, "csvr", "test")

RumexNCTmap5 <- read.cross("csvr", ".", "test.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
#checkAlleles(RumexNCTmap5)


#Switch alleles in a loop
for (i in seq(30, 2, by = -1)) {
  print(i)
  map <- switchAlleles(map, markernames(map, chr=row.names(summaryMap(map)[summaryMap(map)$n.mar<i,])))
  map <- mstmap.cross(map, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-18)
}




