#install.packages("ASMap")
#install.packages("qtl2", repos="hNCTp://rqtl.org/qtl2cran")
#install.packages("qtl")
                 
library(ASMap)
library(qtl)
library(dplyr)
library(tidyr)

setwd("E:/Users/Joanna/Dropbox/Professional/University_of_Toronto/Genomics/HiCSNPs/TranscriptomeLinkageMap/ASMAP/NC_transcriptome/")
setwd("D:/Dropbox/Dropbox/Professional/University_of_Toronto/Genomics/HiCSNPs/TranscriptomeLinkageMap/ASMAP/NC_transcriptome")
load(".RData")
load(".Rhistory")
######-------------- Import and modify data frame from 012 conversion pipeline ------
NCTfile<-read.csv("PosIndvAddedCopyNC_GQ50_0.05_correct_ind_only.csvr", header=T, stringsAsFactors = F)
markers<-as.data.frame(NCTfile$Genotype)
#str(markers)
#markers<-as.data.frame(markers)
colnames(markers)<-"Genotype"
head(markers)
split_markers<-separate(markers, Genotype, c("ScnbKXS","Scaff", "HRSCAF", "HRSCAFnum","Pos"), remove=FALSE)
head(split_markers)

biggest_hundred_scaffolds<-read.table("biggest_hundred_scaffolds.txt")
head(biggest_hundred_scaffolds)
str(split_markers)
str(biggest_hundred_scaffolds)
#intersection<-(as.integer(split_markers$Scaff) %in% biggest_hundred_scaffolds$V1)
sum(as.numeric(intersection))


split_markers$intersect<-(as.integer(split_markers$Scaff) %in% biggest_hundred_scaffolds$V1)
split_markers$intersect[1]<-TRUE #Need to do this to keep vital spacer row for ASMAP format

NCTfile<-left_join(NCTfile, split_markers, by="Genotype")
colnames(NCTfile)
head(NCTfile)
NCTfile<-subset(NCTfile, NCTfile$intersect==T)
NCTfile<-NCTfile[,1:95]



colnames(NCTfile)<-c("Genotype","","sampleA10.NC75M","sampleA11.NC85F","sampleA12.NC93M","sampleA1.NC1F","sampleA2.NC9F","sampleA3.NC17F","sampleA5.NC35F","sampleA6.NC43F","sampleA7.NC51M","sampleA8.NC59M","sampleA9.NC67F","sampleB10.NC77F","sampleB11.NC86F","sampleB12.NC94M","sampleB1.NC2M","sampleB2.NC10F","sampleB3.NC18F","sampleB4.NC27M","sampleB5.NC36M","sampleB6.NC44F","sampleB7.NC52M","sampleB8.NC60F","sampleB9.NC68M","sampleC10.NC78F","sampleC11.NC87F","sampleC12.NC95M","sampleC1.NC3M","sampleC2.NC11F","sampleC3.NC19F","sampleC4.NC28F","sampleC5.NC37F","sampleC6.NC45F","sampleC8.NC61F","sampleC9.NC69F","sampleD10.NC79F","sampleD11.NC88F","sampleD12.NC96F","sampleD1.NC4M","sampleD2.NC12M","sampleD3.NC20M","sampleD4.NC29F","sampleD5.NC38F","sampleD6.NC46M","sampleD8.NC62M","sampleD9.NC70F","sampleE10.NC80M","sampleE11.NC89M","sampleE12.NC97M","sampleE1.NC5F","sampleE2.NC13M","sampleE3.NC21F","sampleE4.NC30F","sampleE5.NC39F","sampleE6.NC47M","sampleE7.NC55F","sampleE8.NC63F","sampleE9.NC71F","sampleF10.NC81F","sampleF11.NC90F","sampleF12.NC98F","sampleF1.NC6M","sampleF2.NC14F","sampleF3.NC22F","sampleF4.NC31F","sampleF5.NC40F","sampleF6.NC48M","sampleF7.NC56F","sampleF8.NC64M","sampleF9.NC72M","sampleG10.NC83M","sampleG11.NC91M","sampleG12.NC99M","sampleG1.NC7M","sampleG2.NC15F","sampleG3.NC23F","sampleG4.NC33M","sampleG5.NC41M","sampleG6.NC49F","sampleG7.NC57F","sampleG8.NC65M","sampleG9.NC73M","sampleH10.NC84M","sampleH11.NC92F","sampleH12.NC100F","sampleH1.NC8F","sampleH2.NC16F","sampleH3.NC24F","sampleH4.NC34M","sampleH5.NC42F","sampleH6.NC50M","sampleH7.NC58F","sampleH8.NC66F","sampleH9.NC74M")
head(NCTfile)
ncol(NCTfile)
NCTfile[1,2]<-""
write.csv(NCTfile, "ModifiedGenotypeFile.csvr", row.names = F) #CSVR file for input into ASMAP
#warnings()
#dim(NCTfile)[1]
rm(NCTfile)
rm(markers)
rm(split_markers)
########### inspect and filter raw data ##### ---------
RumexNCTmap <- read.cross("csvr", ".", "ModifiedGenotypeFile.csvr", crosstype = "f2",genotypes=c("A","H","B"), F.gen = 2)
plotMissing(RumexNCTmap) #A couple of individuals with high missing data, most not too bad
?read.cross
?convert2bcsft
RumexNCTmap <-convert2bcsft(RumexNCTmap, F.gen = 2, BC.gen=0,estimate.map = FALSE)
#For a population that has been generated by selfing n times the conversion function convert2bcsft can be used by seNCTing the arguments F.gen = n and BC.gen = 0
plotMissing(RumexNCTmap1) #A couple of individuals with high missing data, most not too bad

#Remove missing data ---------------
sg <- statGen(RumexNCTmap, bychr = FALSE, stat.type = "miss")
hist(sg$miss)
RumexNCTmap1 <- subset(RumexNCTmap, ind = sg$miss < 5000)
#Check for clones
gc <- genClones(RumexNCTmap1, tol = 0.95)
gc$cgd
cgd <- gc$cgd
RumexNCTmap2 <- fixClones(RumexNCTmap1, cgd, consensus = TRUE)

levels(RumexNCTmap2$pheno[[1]])[grep("_", levels(RumexNCTmap2$pheno[[1]]))]
#One pair look like clones - match at 0.9997 of sites

#Remove co-located markers ------------------------------

RumexNCTmap3 <- pullCross(RumexNCTmap2, type = "co.located")
names(RumexNCTmap3)
sum(ncol(RumexNCTmap3$seg.dist$data),ncol(RumexNCTmap3$co.located$data))
?pullCross
str(RumexNCTmap3$co.located$data)
RumexNCTmap3$co.located
View(RumexNCTmap3$co.located$table)
write.csv(RumexNCTmap3$co.located$table, "Colocated_markers_biggest_100.csv")


#Construct the map ---------------------
RumexNCTmap4 <- mstmap(RumexNCTmap3, bychr = FALSE, trace = TRUE, dist.fun ="kosambi", p.value = 1e-11)
heatMap(RumexNCTmap4)
#?mstmap
nmar(RumexNCTmap4)
chrlen(RumexNCTmap4)
length(chrlen(RumexNCTmap4) )

save.image()
#Fix switched alleles - some alleles switched because parental genotypes not known therefore data not phased
#The easiest way to do this is to export the map, import it again as an F2 map, run checkalleles and switch alleles, and then put it back into ASMAP
#Use loop to switch the alleles in all the linkage groups in sequence

rm(RumexNCTmap);rm(RumexNCTmap1);rm(RumexNCTmap2);rm(RumexNCTmap3)

write.cross(RumexNCTmap4, "csvr", "test")
#Loop to switch alleles
for (i in seq(2, 60, by = 1)){
  print(c("i = ", i))
  print(length(nmar(RumexNCTmap4)))
  if(i > length(nmar(RumexNCTmap4))){
    print("DONE")}
  else {
    RumexNCTmap4 <- read.cross("csvr", ".", "test.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
    num_markers<-data.frame(name = names(nmar(RumexNCTmap4)), num_mar = nmar(RumexNCTmap4))
    current_chr<-num_markers[order(-num_markers$num_mar),]$name[i]
    print("Current chromosome")
    print(current_chr)
    print("Number of markers")
    print(num_markers[order(-num_markers$num_mar),]$num_mar[i])
    nmar(RumexNCTmap4)
    #plotMap(RumexNCTmap4, chr=i)
    RumexNCTmap4 <- switchAlleles(RumexNCTmap4, markernames(RumexNCTmap4, chr=(paste("L.",i, sep=""))))
    print("Alleles switched")
    RumexNCTmap4 <-convert2bcsft(RumexNCTmap4, F.gen = 2, BC.gen=0,estimate.map = FALSE)
    print("convert2bcsft")
    RumexNCTmap4 <- mstmap(RumexNCTmap4, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-11)
    print("mstmap")
    RumexNCTmap4
    write.cross(RumexNCTmap4, "csvr", "test")
    print("written")
  }
}
#Run loop three times, and manually switch any LG that has the same number of markers as another
save.image()
load.image()
nmar(RumexNCTmap4)
heatMap(RumexNCmap4)
plotMap(RumexNCTmap4)


RumexNCTmap4 <- mstmap(RumexNCTmap4, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-11)

#
#Inspect markers
#Drop markers with >0.8 heterozygous genotypes, because those are almost certainly mapping errors

#statMark(RumexNCTmap5, stat.type="marker")$marker$AB>0.8
mm <- statMark(RumexNCTmap4, stat.type = "marker")$marker$AB
View(mm)
sum(as.numeric(mm))
RumexNCTmap4 <- drop.markers(RumexNCTmap4, c(markernames(RumexNCTmap4)[mm > 0.8]))

RumexNCTmap5<-pullCross(RumexNCTmap4, type = "seg.distortion", pars =
                         list(seg.thresh = "bonf")) #Pull distorted markers for inspection

write.csv(RumexNCTmap5$seg.distortion$table, "distorted_markers_biggest_100.csv")

RumexNCTmap5 <- mstmap(RumexNCTmap5, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-5)
heatMap(RumexNCTmap5)
#Clearly one LG still needs to be switched
write.cross(RumexNCTmap5, "csvr", "test")
RumexNCTmap5 <- read.cross("csvr", ".", "test.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
RumexNCTmap5 <- switchAlleles(RumexNCTmap5, markernames(RumexNCTmap5, chr=(paste("L.",10, sep=""))))
RumexNCTmap5 <-convert2bcsft(RumexNCTmap5, F.gen = 2, BC.gen=0,estimate.map = FALSE)
RumexNCTmap5 <- mstmap(RumexNCTmap5, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-5)



nmar(RumexNCTmap5)
chrlen(RumexNCTmap5)
plot.map(RumexNCTmap5)
pull.map(RumexNCTmap5)


#Check crossovers, double recombinants, and missing data and remove problematic individuals --------------

pg <- profileGen(RumexNCTmap5, bychr = FALSE, stat.type = c("xo", "dxo",
                                                           "miss"), id = "Genotype", xo.lambda = 10, layout = c(1, 3), lty = 2, cex =   0.7)

pg$xo.lambda
pgframe<-as.data.frame(pg$xo.lambda)
subset(pgframe, pg$xo.lambda==T)
RumexNCTmap6 <- subsetCross(RumexNCTmap5, ind = !pg$xo.lambda) #Remove individuals with high double crossovers
RumexNCTmap6 <- mstmap(RumexNCTmap6, bychr = TRUE, dist.fun = "kosambi", trace = TRUE,
                       p.value = 1e-4)
nmar(RumexNCTmap6)
chrlen(RumexNCTmap6)
heatMap(RumexNCTmap6)
pull.map(RumexNCTmap6)
heatMap(RumexNCTmap6,chr = c("L.2","L.3"))
heatMap(RumexNCTmap6,chr = c("L.4","L.5"))
heatMap(RumexNCTmap6,chr = c("L.1","L.6"))
heatMap(RumexTTmap6,chr = c("L.10","L.11"))
library(ASMap)
heatMap(RumexNCTmap8,chr = c("L.1","L.6"))
plot.map(RumexNCTmap6)
load(".RData")
profileMark(RumexNCTmap6, stat.type = c("seg.dist", "prop", "dxo", "recomb"),
            layout = c(1, 5), type = "l")

write.cross(RumexNCTmap6, "csvr", "F2_style_NC_map_clean_biggest100_5-28")