#install.packages("ASMap")
#install.packages("qtl2", repos="http://rqtl.org/qtl2cran")
#install.packages("qtl")
#install.packages("stingr")
#install.packages("tidyr")

#library(stingr)
library(ASMap)
library(tidyr)
library(qtl)
library(dplyr)

#setwd("E:/Users/Joanna/Dropbox/Professional/University_of_Toronto/Genomics/HiCSNPs/TranscriptomeLinkageMap/ASMAP/TX_transcriptome/")
#setwd("D:/Dropbox/Dropbox/Professional/University_of_Toronto/Genomics/HiCSNPs/TranscriptomeLinkageMap/ASMAP/TX_transcriptome")
setwd("D:/Dropbox/Dropbox/Professional/University_of_Toronto/Genomics/Rumex_genome_paper/Finalized_analyses")
load(".RData")
load(".Rhistory")
######-------------- Import and modify data frame from 012 conversion pipeline ------
TTfile<-read.csv("D:/Dropbox/Dropbox/Professional/University_of_Toronto/Genomics/HiCSNPs/TranscriptomeLinkageMap/ASMAP/TX_transcriptome/PosIndvAddedCopyTX_GQ50_0.05_correct_ind_only.csvr", header=T, stringsAsFactors = F) #Import dataset formatted as CSVR
markers<-as.data.frame(TTfile$Genotype) #Collect markers to filter by scaffold
colnames(markers)<-"Genotype"
split_markers<-separate(markers, Genotype, c("ScnbKXS","Scaff", "HRSCAF", "HRSCAFnum","Pos"), remove=FALSE)

biggest_hundred_scaffolds<-read.table("D:/Dropbox/Dropbox/Professional/University_of_Toronto/Genomics/HiCSNPs/TranscriptomeLinkageMap/ASMAP/TX_transcriptome/sex_linked_and_biggest_hundred_scaffolds.txt")

split_markers$intersect<-(as.integer(split_markers$Scaff) %in% biggest_hundred_scaffolds$V1) #Identify subset of markers in biggest hundred scaffolds plus sex linked scaffolds of interest
split_markers$intersect[1]<-TRUE #Need to do this to keep vital spacer row

TTfile<-left_join(TTfile, split_markers, by="Genotype")
TTfile<-subset(TTfile, TTfile$intersect==T)
TTfile<-TTfile[,1:96]

colnames(TTfile)<-c("Genotype","","sampleA10.TX76F","sampleA11.TX84M","sampleA12.TX93F","sampleA1.TX1F","sampleA2.TX9F","sampleA3.TX17F","sampleA4.TX25M","sampleA5.TX33M","sampleA6.TX42M","sampleA7.TX51F","sampleA8.TX59F","sampleA9.TX68M","sampleB10.TX77M","sampleB11.TX86F","sampleB12.TX94M","sampleB1.TX2M","sampleB2.TX10F","sampleB3.TX18M","sampleB4.TX26M","sampleB5.TX34M","sampleB6.TX43M","sampleB7.TX52F","sampleB8.TX61M","sampleB9.TX69F","sampleC10.TX78M","sampleC11.TX87F","sampleC12.TX95F","sampleC1.TX3M","sampleC2.TX11F","sampleC3.TX19M","sampleC4.TX27M","sampleC5.TX35F","sampleC6.TX44M","sampleC7.TX53F","sampleC8.TX62F","sampleC9.TX70M","sampleD10.TX79F","sampleD11.TX88M","sampleD12.TX96F","sampleD1.TX4M","sampleD2.TX12F","sampleD3.TX20F","sampleD4.TX28M","sampleD5.TX36M","sampleD6.TX45M","sampleD7.TX54M","sampleD8.TX63F","sampleD9.TX71M","sampleE10.TX80M","sampleE11.TX89F","sampleE12.TX97M","sampleE1.TX5F","sampleE2.TX13M","sampleE3.TX21M","sampleE4.TX29M","sampleE5.TX37M","sampleE6.TX46M","sampleE7.TX55F","sampleE8.TX64F","sampleE9.TX72M","sampleF10.TX81F","sampleF11.TX90M","sampleF12.TX98F","sampleF1.TX6F","sampleF2.TX14F","sampleF3.TX22F","sampleF4.TX30F","sampleF5.TX38M","sampleF6.TX47M","sampleF7.TX56M","sampleF8.TX65M","sampleF9.TX73F","sampleG10.TX82F","sampleG11.TX91F","sampleG12.TX99M","sampleG1.TX7M","sampleG2.TX15F","sampleG3.TX23F","sampleG4.TX31M","sampleG5.TX39F","sampleG6.TX49M","sampleG7.TX57F","sampleG8.TX66F","sampleG9.TX74M","sampleH11.TX92M","sampleH12.TX100M","sampleH1.TX8F","sampleH2.TX16F","sampleH3.TX24M","sampleH4.TX32M","sampleH5.TX40M","sampleH6.TX50M","sampleH7.TX58M","sampleH8.TX67M")

TTfile[1,2]<-""

write.csv(TTfile, "ModifiedGenotypeFile_sex_linked_plus_biggest_hundred_scaffolds.csvr", row.names = F) 

rm(TTfile)
rm(markers)
rm(split_markers)


########### inspect and filter raw data ##### ---------
RumexTTmap <- read.cross("csvr", ".", "ModifiedGenotypeFile_sex_linked_plus_biggest_hundred_scaffolds.csvr", crosstype = "f2",genotypes=c("A","H","B"), F.gen = 2)
plotMissing(RumexTTmap) #A couple of individuals with high missing data, most not too bad
RumexTTmap <-convert2bcsft(RumexTTmap, F.gen = 2, BC.gen=0,estimate.map = FALSE)
#For a population that has been generated by selfing n times the conversion function convert2bcsft can be used by setting the arguments F.gen = n and BC.gen = 0

#Remove missing data ---------------
sg <- statGen(RumexTTmap, bychr = FALSE, stat.type = "miss")
hist(sg$miss)
RumexTTmap1 <- subset(RumexTTmap, ind = sg$miss < 5000)
plotMissing(RumexTTmap1) #A couple of individuals with high missing data, most not too bad
#Check for clones
gc <- genClones(RumexTTmap1, tol = 0.95)
gc$cgd
cgd <- gc$cgd
RumexTTmap2 <- fixClones(RumexTTmap1, cgd, consensus = TRUE)

levels(RumexTTmap2$pheno[[1]])[grep("_", levels(RumexTTmap2$pheno[[1]]))]
#One pair look like clones - match at 0.9997 of sites

#Remove co-located markers ------------------------------

RumexTTmap3 <- pullCross(RumexTTmap2, type = "co.located")
sum(ncol(RumexTTmap3$seg.dist$data),ncol(RumexTTmap3$co.located$data))
#str(RumexTTmap3$co.located$data)
#RumexTTmap3$co.located
#View(RumexTTmap3$co.located$table)
write.csv(RumexTTmap3$co.located$table, "Colocated_markers_biggest_100_plus_sex_linked.csv")
RumexTTmap$pheno

#Construct the map ---------------------
RumexTTmap4 <- mstmap(RumexTTmap3, bychr = FALSE, trace = TRUE, dist.fun ="kosambi", p.value = 1e-11)
heatMap(RumexTTmap4)
#?mstmap
nmar(RumexTTmap4)
chrlen(RumexTTmap4)
length(chrlen(RumexTTmap4) )


#Fix switched alleles --------------------
#The easiest way to do this is to export the map, import it again as an F2 map, run checkalleles and switch alleles, and then put it back into ASMAP
#Use Kate's loop to switch the alleles in all the linkage groups in sequence

write.cross(RumexTTmap4, "csvr", "test")
i #Loop to switch alleles
for (i in seq(2, 60, by = 1)){
  print(c("i = ", i))
  print(length(nmar(RumexTTmap4)))
  if(i > length(nmar(RumexTTmap4))){
    print("DONE")}
  else {
    RumexTTmap4 <- read.cross("csvr", ".", "test.csv", crosstype = "f2",genotypes=c("AA","AB","BB"), F.gen = 2)
    num_markers<-data.frame(name = names(nmar(RumexTTmap4)), num_mar = nmar(RumexTTmap4))
    current_chr<-num_markers[order(-num_markers$num_mar),]$name[i]
    print("Current chromosome")
    print(current_chr)
    print("Number of markers")
    print(num_markers[order(-num_markers$num_mar),]$num_mar[i])
    nmar(RumexTTmap4)
    #plotMap(RumexTTmap4, chr=i)
    RumexTTmap4 <- switchAlleles(RumexTTmap4, markernames(RumexTTmap4, chr=(paste("L.",i, sep=""))))
    print("Alleles switched")
    RumexTTmap4 <-convert2bcsft(RumexTTmap4, F.gen = 2, BC.gen=0,estimate.map = FALSE)
    print("convert2bcsft")
    RumexTTmap4 <- mstmap(RumexTTmap4, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-11)
    print("mstmap")
    RumexTTmap4
    write.cross(RumexTTmap4, "csvr", "test")
    print("written")
  }
}
RumexTTmap4$pheno
?mstmap
#Run loop twice, and manually switch any LG that has the same number of markers as another - in this case, only linkage groups with under 10 markers fit that criterion so skipped

save.image()
load.image()

View(num_markers)
nmar(RumexTTmap4)
heatMap(RumexTTmap4)

plotMap(RumexTTmap4)


RumexTTmap4 <- mstmap(RumexTTmap4, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-11)

#Inspect segregation distortion and pull distorted markers -------
#Inspect markers
pm<-profileMark(RumexTTmap4, stat.type = c("seg.dist", "prop", "dxo", "recomb"),
                layout = c(1, 5), type = "l")
mm <- statMark(RumexTTmap4, stat.type = "marker")$marker$AB
RumexTTmap4 <- drop.markers(RumexTTmap4, c(markernames(RumexTTmap4)[mm > 0.8])) #Drop markers with >0.8 heterozygous genotypes, because those are almost certainly mapping errors
RumexTTmap5<-pullCross(RumexTTmap4, type = "seg.distortion", pars =
                          list(seg.thresh = "bonf")) #Pull distorted markers for inspection
write.csv(RumexTTmap5$seg.distortion$table, "distorted_markers_100_biggest_scaffolds_plus_sex_linked.csv")



RumexTTmap5 <- mstmap(RumexTTmap5, id="NA.", bychr=FALSE, dist.fun="kosambi", trace=TRUE, p.value=1e-6)
heatMap(RumexTTmap5)
nmar(RumexTTmap5)
chrlen(RumexTTmap5)
plot.map(RumexTTmap5)
pull.map(RumexTTmap5)


#Check crossovers, double recombinants, and missing data and remove problematic individuals --------------

pg <- profileGen(RumexTTmap5, bychr = FALSE, stat.type = c("xo", "dxo",
                                                           "miss"), id = "Genotype", xo.lambda = 10, layout = c(1, 3), lty = 2, cex =   0.7)

pg$xo.lambda
pgframe<-as.data.frame(pg$xo.lambda)
subset(pgframe, pg$xo.lambda==T)
RumexTTmap6 <- subsetCross(RumexTTmap5, ind = !pg$xo.lambda)
RumexTTmap6 <- mstmap(RumexTTmap6, bychr = F, dist.fun = "kosambi", trace = TRUE,
                       p.value = 1e-5)
nmar(RumexTTmap6)
chrlen(RumexTTmap6)
heatMap(RumexTTmap6)
plotMap(RumexTTmap6)
heatMap(RumexTTmap6,chr = c("L.5", "L.6"))
heatMap(RumexTTmap6,chr = c("L.7", "L.11"))
heatMap(RumexTTmap6,chr = c("L.5", "L.10"))
heatMap(RumexTTmap6,chr = c("L.5", "L.3"))


write.cross(RumexTTmap6, "csvr", "Final_F2_style_TX_map_biggest_100_scaffolds_plus_sex_linked_10-21") 
pull.map(RumexTTmap6)
save.image()
